app/queries/arel_helpers.rb

```diff
@@ -0,0 +1,34 @@
+module Queries
+  module ArelHelpers
+    CTE = Struct.new(:table, :composed_cte)
+
+    def table(table)
+      Arel::Table.new(table, ActiveRecord::Base)
+    end
+
+    def cte(scope, as:)
```

Comment:
Tried, couldn't really make it any nicer. I added a whopping great comment. Let me know if it makes sense or not, or if anyone has any bright ideas for a nicer abstraction for CTEs